Resolve-Dependency.ps1

```diff
@@ -285,48 +287,82 @@ try
     if (-not $powerShellGetVersion -or ($powerShellGetVersion -lt [System.Version] '2.0' -and -not $AllowOldPowerShellGetModule))
     {
-        Write-Progress -Activity 'Bootstrap:' -PercentComplete 40 -CurrentOperation 'Installing newer version of PowerShellGet'
-
-        $installPowerShellGetParameters = @{
-            Name               = 'PowerShellGet'
-            Force              = $True
-            SkipPublisherCheck = $true
-            AllowClobber       = $true
-            Scope              = $Scope
-            Repository         = $Gallery
-        }
+        Write-Progress -Activity 'Bootstrap:' -PercentComplete 40 -CurrentOperation 'Fetching newer version of PowerShellGet'
 
-        switch ($PSBoundParameters.Keys)
+        # PowerShellGet module not found, installing or saving it.
+        if ($PSDependTarget -in 'CurrentUser', 'AllUsers')
         {
-            'Proxy'
-            {
-                $installPowerShellGetParameters.Add('Proxy', $Proxy)
-            }
+            Write-Debug -Message "PowerShellGet module not found. Attempting to install from Gallery $Gallery."
 
-            'ProxyCredential'
-            {
-                $installPowerShellGetParameters.Add('ProxyCredential', $ProxyCredential)
+            Write-Warning -Message "Installing PowerShellGet in $PSDependTarget Scope."
+
+            $installPowerShellGetParameters = @{
+                Name               = 'PowerShellGet'
+                Force              = $true
+                SkipPublisherCheck = $true
+                AllowClobber       = $true
+                Scope              = $Scope
+                Repository         = $Gallery
             }
 
-            'GalleryCredential'
+            switch ($PSBoundParameters.Keys)
             {
-                $installPowerShellGetParameters.Add('Credential', $GalleryCredential)
+                'Proxy'
+                {
+                    $installPowerShellGetParameters.Add('Proxy', $Proxy)
+                }
+
+                'ProxyCredential'
+                {
+                    $installPowerShellGetParameters.Add('ProxyCredential', $ProxyCredential)
+                }
+
+                'GalleryCredential'
+                {
+                    $installPowerShellGetParameters.Add('Credential', $GalleryCredential)
+                }
             }
 
-            'AllowPrerelease'
-            {
-                $installPowerShellGetParameters.Add('AllowPrerelease', $AllowPrerelease)
+            Write-Progress -Activity 'Bootstrap:' -PercentComplete 60 -CurrentOperation 'Installing newer version of PowerShellGet'
+
+            Install-Module @installPowerShellGetParameters
+        }
+        else
+        {
+            Write-Debug -Message "PowerShellGet module not found. Attempting to Save from Gallery $Gallery to $PSDependTarget"
+
+            $saveModuleParameters = @{
+                Name           = 'PowerShellGet'
+                MaximumVersion = '2.99.99'
```

Comment:
Why?
Since PowerShellGet v3 is only pre-release there's no arm in removing this until proven it breaks??
```suggestion
```

Reply:
Removed. I was dived by adding `MaximumVersion`. I thought v3 won't work since there are new commands in that version and needed to be added new code to support v3 anyway, so by adding `MaximumVersion` the code would work even if v3 also was installed.